File: src/main/scala/codility/lessons/prefixSum/PassingCars.scala
package codility.lessons.prefixSum

/**
 * Level : painless
 */
object PassingCarsScala {
    object Solution {
        private val MAX = 1000000000
        
        def solution(arr: Array[Int]): Int = {
            var (eastCars, passingCars) = (0, 0)
            for (car <- arr) {
                if (car == 0) eastCars += 1
                else  { 
                    passingCars += eastCars
                    if (passingCars > MAX) return -1
                }
            }
            passingCars 
        }
    }
}

File: src/main/java/codility/lessons/prefixSum/PassingCars.java
package codility.lessons.prefixSum;

/**
 * Level : painless
 */
public class PassingCars {
	public int solution(int[] A) {
		int west = 0;
		for (int i = 1; i < A.length; i++)
			west += A[i];

		int count = 0;
		for (int i = 0; i < A.length - 1; i++) {
			if (A[i] == 0)
				count += west;
			else
				west--;
			if (count > 1_000_000_000)
				return -1;
		}
		return count;
	}
}
